packages/core/src/domain/configuration.ts

```diff
@@ -68,6 +68,8 @@ export interface UserConfiguration {
   replica?: ReplicaUserConfiguration
 }
 
+export type PublicUserConfiguration = Pick<UserConfiguration, 'service' | 'env' | 'version'>
```

Comment:
> I intentionally published only a subset of the user configuration - otherwise, a third-party script could mutate user configuration which could have some security implications 🤔

Accessing the whole user configuration could probably be helpful for other uses cases, what about returning a copy of it to avoid any mutation?

Reply:
Do you mean deep clone? Because there are some complex values, like arrays, regexp, objects.
In my opinion it make sense to limit API as much as possible and extend it when there is a request for it. Adding new fields to this type will not be a breaking change, but removing will be :)

Reply:
> Do you mean deep clone? Because there are some complex values, like arrays, regexp, objects.

👍

> In my opinion it make sense to limit API as much as possible and extend it when there is a request for it. Adding new fields to this type will not be a breaking change, but removing will be :)

I get your point but accessing all values would already be helpful while debugging (easier than find where the SDK is init) or if we want to retrieve the whole configuration from a browser extension.

Reply:
I was trying to deepClone using built-in function from `context.ts` but unfortunately it doesn't support RegExp and functions which can be a part of the init object. Should we omit these values or just copy references to them?
It also reports a type-error even if InitConfiguration is compatible with `ContextValue`. This issue can be solved by upgrading TypeScript to at least 4.2.3.

Reply:
>  Should we omit these values or just copy references to them?

no way to clone them? (IE11 compatibility)
otherwise, copy the reference should be ok IMO

> It also reports a type-error even if InitConfiguration is compatible with ContextValue.

Do you have more details on that?

Reply:
![image](https://user-images.githubusercontent.com/8374473/122771099-65d82400-d2a6-11eb-948d-7ea277623533.png)
I was wrong - upgrading TypeScript didn't help :/ It seems that TypeScript is missing index signature from the `InitConfiguration` type

Reply:
if we extend `deepClone` to handle (somehow) other types than the those supported by `Context`, it could worth removing the type limitation and move `deepClone` to `core/src/tools/utils`.

Reply:
I updated the PR with generic deepClone, mergeInto and combine :)